tests/views/disabled_project.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import pytest
+
+from django.core.urlresolvers import reverse
+
+
+@pytest.mark.django_db
+def test_views_disabled_project(client, dp_view_urls, request_users):
+    url = dp_view_urls
+    user = request_users["user"]
+    if user.username != "nobody":
+        client.login(
+            username=user.username,
+            password=request_users["password"])
+
+    response = client.get(url)
+
+    if user.is_superuser:
+        assert response.status_code == 200
+    else:
+        assert response.status_code == 404
+
+
+@pytest.mark.django_db
+def test_disabled_project_in_lang_browse_view(client, request_users):
+    user = request_users["user"]
+    if user.username != "nobody":
+        client.login(
+            username=user.username,
+            password=request_users["password"])
+
+    response = client.get(reverse("pootle-language-browse",
+                                  kwargs={"language_code": "language0"}))
+
+    disabled_project_exists = "language0-disabled_project0" in [
+        item["code"] for item in response.context["table"]["items"]]
+
+    assert not(user.is_superuser ^ disabled_project_exists)
```

Comment:
Is this a "dear lazy web"? https://docs.python.org/2/library/operator.html#mapping-operators-to-functions.  Though I think `^=` gives what is wanted. Better, I think, to ask these on the channel.